DeveloperLabs/MyEmployees/MyEmployees/Form1.cs

```diff
@@ -32,10 +36,80 @@ private void Form1_Load(object sender, EventArgs e)
         {
             LoadConfig();
             LoadData();
+            AddAppInstaller(getAppInstallerUri());
             CheckKioskMode();
             //await CheckForUpdates();
         }
 
+        // Retrieves the URI of the .appinstaller which is embedded in the application (MyEmployees) package
+        public static string getAppInstallerUri()
+        {
+            string uriPath = null;
+            try
+            {
+                uriPath = System.IO.Path.GetDirectoryName(System.Windows.Forms.Application.ExecutablePath) + "\\..\\Update.appinstaller";
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+            return uriPath;
+        }
+
+        // If there is not an associated .appinstaller file with the package,
+        // this function adds the .appinstaller referred to by inputPackageUri
+        public static void AddAppInstaller(String inputPackageUri)
+        {
+            AppInstallerInfo info = Package.Current.GetAppInstallerInfo();
+            if (info == null && inputPackageUri != null)
+            {
+
+                Uri packageUri = new Uri(inputPackageUri);
+                PackageManager packageManager = new PackageManager();
+                IAsyncOperationWithProgress<DeploymentResult, DeploymentProgress> deploymentOperation = null;
+                try
+                {
+                    deploymentOperation = packageManager.AddPackageByAppInstallerFileAsync(
+                            packageUri,
+                            AddPackageByAppInstallerOptions.ForceTargetAppShutdown,
+                            null);
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
+                }
+
+                // Register the active instance of an application for restart
+                uint res = RelaunchHelper.RegisterApplicationRestart(null, RelaunchHelper.RestartFlags.NONE);

```

Comment:
Should this restart registration occur before the AddPackageByAppIsntallerFileAsync call to ensure they are not racing? IE If the AddPackageByAppIsntallerFileAsync restarted the application before the RegisterApplicationRestart was complete, it might not restart?

Reply:
Yep, good point. I didn't realize because I wasn't getting any issues while testing but this has been addressed in latest commit.